tmt/schemas/common.yaml

```diff
@@ -0,0 +1,137 @@
+---
+
+#
+# Common JSON Schema definitions
+#
+
+$id: /schemas/common
+$schema: https://json-schema.org/draft-07/schema
+
+definitions:
+  # https://tmt.readthedocs.io/en/stable/spec/tests.html#require
+  beakerlib_library:
+    allOf:
+      - $ref: "/schemas/common#/definitions/fmf_id_base"
+
+    additionalProperties: false
+
+    properties:
+      url: true
+      ref: true
+      path: true
+      name: true
+      nick:
+        type: string
+
+  # https://fmf.readthedocs.io/en/stable/concept.html#identifiers
+  fmf_id:
+    allOf:
+      - $ref: "/schemas/common#/definitions/fmf_id_base"
+
+    additionalProperties: false
+
+    properties:
+      url: true
+      ref: true
+      path: true
+      name: true
+
+  # helper used by beakerlib_library and fmf_id
+  fmf_id_base:
+    type: object
+    minProperties: 1
+    properties:
+      url:
+        type: string
+        # TODO: maybe also "format: uri"
+        # together with "pattern" to enforce http instead
+        # of all possible protocols?
+        pattern: "^https?://"
```

Comment:
It can be any format acceptable by `git clone` so for example `git@github.com:teemtee/tmt.git` as well, right?

Reply:
do we want to lock it down or just leave it as string? I believe I could come up with a pattern for possible git URIs

Reply:
I'd be ok with `string`. Btw, I guess we will give warnings for some time before strictly enforcing the spec? Not to break too much stuff ;)

Reply:
ok, let's keep a string, I have filled: https://github.com/teemtee/tmt/issues/1258